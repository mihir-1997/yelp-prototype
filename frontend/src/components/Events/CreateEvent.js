import React, { Component } from 'react'
import axios from 'axios'
import { Redirect } from 'react-router';

import './Events.css'

export default class CreateEvent extends Component {

    constructor( props ) {
        super( props )
        this.state = {
            name: "",
            description: "",
            location: "",
            date: "",
            time: "",
            hashtags: "",
            error: ""
        }
    }

    onChange = item => {
        this.setState( { [ item.target.name ]: item.target.value } );
    }

    postEvent = ( item ) => {
        item.preventDefault()
        if ( !this.state.time.includes( ":" ) || this.state.time.length !== 5 ) {
            this.setState( {
                error: "Time must be in HH:MM format"
            } )
            return
        } else {
            this.setState( {
                error: ""
            } )
        }
        let hour = this.state.time.split( ":" )[ 0 ]
        let minutes = this.state.time.split( ":" )[ 1 ]
        if ( parseInt( hour ) < 1 || parseInt( hour ) > 23 ) {
            console.log( parseInt( hour ) )
            this.setState( {
                error: "Hour must be between 1 to 23"
            } )
            return
        } else if ( parseInt( minutes ) < 0 || parseInt( minutes ) > 59 ) {
            this.setState( {
                error: "Minuted must be between 0 to 59"
            } )
            return
        } else {
            this.setState( {
                error: ""
            } )
            if ( this.state.name && this.state.description && this.state.location && this.state.date && this.state.time ) {
                let id = localStorage.getItem( "id" )
                const eventData = {
                    restaurant_id: id,
                    name: this.state.name,
                    description: this.state.description,
                    location: this.state.location,
                    date: this.state.date,
                    time: this.state.time + ":00",
                    hashtags: this.state.hashtags,
                }
                axios.defaults.withCredentials = true;
                axios.post( "http://localhost:3001/postevent", eventData )
                    .then( ( res ) => {
                        if ( res.status === 200 ) {
                            this.setState( {
                                error: ""
                            } )
                            window.location.assign( "/events" );
                        }
                    } )
                    .catch( ( err ) => {
                        if ( err.response ) {
                            console.log( err.response )
                        }
                    } )
            } else {
                this.setState( {
                    error: "*Some required fields are empty"
                } )
            }
        }
    }

    render () {
        var redirectVar = null
        if ( localStorage.getItem( "active" ) !== "restaurant" ) {
            redirectVar = <Redirect to="/login" />
            return redirectVar
        }
        return (
            <div >
                {redirectVar }
                <div className="container">
                    <div className="row">
                        <div className="col-2"></div>
                        <div className="col-8 create-event-form-wrapper">
                            <div className="row">
                                <div className="text-center">
                                    <h3>Create Event</h3>
                                </div>
                            </div>
                            <form>
                                <div class="form-group">
                                    <label for="name">Name<span className="required-field">*</span></label>
                                    <input type="text" class="form-control" id="name" name="name" placeholder="Event name" value={ this.state.name } onChange={ this.onChange } required />
                                </div>
                                <div class="form-group">
                                    <label for="description">Description<span className="required-field">*</span></label>
                                    <textarea type="text" class="form-control" name="description" id="description" value={ this.state.description } onChange={ this.onChange } required />
                                </div>
                                <div class="form-group">
                                    <label for="location">Location<span className="required-field">*</span></label>
                                    <input type="text" class="form-control" id="location" name="location" value={ this.state.location } onChange={ this.onChange } required />
                                </div>
                                <div class="form-group">
                                    <label for="hashtags">Hashtags</label>
                                    <input type="text" class="form-control" id="hashtags" name="hashtags" placeholder="#tasty" value={ this.state.hashtags } onChange={ this.onChange } />
                                </div>
                                <div class="form-row">
                                    <div class="form-group col-md-6">
                                        <label for="date">Date<span className="required-field">*</span></label>
                                        <input type="date" class="form-control" id="date" name="date" value={ this.state.date } min={ new Date().toISOString().split( "T" )[ 0 ] } onChange={ this.onChange } required />
                                    </div>
                                    <div class="form-group col-md-6">
                                        <label for="time">Time<span className="required-field">*</span></label>
                                        <input type="text" class="form-control" id="time" name="time" value={ this.state.time } placeholder="HH:MM" onChange={ this.onChange } required />
                                    </div>
                                </div>
                                <button type="submit" class="btn red-button" onClick={ this.postEvent }>Create</button>
                            </form>
                            <p className="error">{ this.state.error }</p>
                        </div>
                        <div className="col-2"></div>
                    </div>
                </div>
            </div>
        )
    }
}
