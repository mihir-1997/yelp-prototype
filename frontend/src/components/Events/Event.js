import React, { Component } from 'react'
import axios from 'axios'

export default class Event extends Component {

    constructor( props ) {
        super( props )
        this.state = {
            id: this.props.event.id,
            name: this.props.event.name,
            user_id: this.props.event.user_id,
            restaurant_id: this.props.event.restaurant_id,
            restaurant_name: this.props.event.restaurant_name,
            user_ids: this.props.event.user_ids,
            user_names: this.props.event.user_names,
            description: this.props.event.description,
            location: this.props.event.location,
            date: this.props.event.date,
            time: this.props.event.time,
            hashtags: this.props.event.hashtags,
            registered: this.props.registered
        }
    }

    registerForEvent = ( item ) => {
        item.preventDefault()
        let id = localStorage.getItem( "id" )
        if ( id ) {
            const events_sub_data = {
                user_id: id,
                event_id: this.state.id
            }
            axios.post( "http://localhost:3001/registeForEvent", events_sub_data )
                .then( ( res ) => {
                    if ( res.status === 200 ) {
                        window.location.reload()
                    }
                } )
                .catch( ( err ) => {
                    if ( err.response ) {
                        console.log( err.response.message )
                        return
                    }
                    return
                } )
        } else {
            console.log( "No Id found in local storage" )
        }
    }

    showUserProfile = ( user_id ) => {
        this.props.showprofile( user_id )
    }

    render () {
        let column = null
        let event_name = null
        if ( this.props.registered ) {
            column = "col-12"
            event_name = <h4 className="event-name">{ this.props.event.name }</h4>
        } else {
            column = "col-9"
            event_name = <h3 className="event-name">{ this.props.event.name }</h3>
        }
        return (
            <div className="container">
                <div className="row event">
                    <div className={ column }>
                        <div className="row">
                            { event_name }
                        </div>
                        <div className="row">
                            <div className="col">
                                { localStorage.getItem( "active" ) === "user" &&
                                    <span>
                                        <strong>{ this.props.event.restaurant_name }</strong>
                                        <br />
                                    </span>
                                }
                                <svg viewBox="0 0 15 15" fill="none" xmlns="http://www.w3.org/2000/svg" width="15" height="15"><path d="M7.5.5v14m7-7.005H.5m13 0a6.006 6.006 0 01-6 6.005c-3.313 0-6-2.694-6-6.005a5.999 5.999 0 016-5.996 6 6 0 016 5.996z" stroke="currentColor" stroke-linecap="square"></path></svg>
                                    &nbsp;&nbsp;{ this.props.event.location }
                                <br />
                                <svg viewBox="0 0 15 15" fill="none" xmlns="http://www.w3.org/2000/svg" width="15" height="15"><path d="M3.5 0v5m8-5v5m-10-2.5h12a1 1 0 011 1v10a1 1 0 01-1 1h-12a1 1 0 01-1-1v-10a1 1 0 011-1z" stroke="currentColor"></path></svg>
                                    &nbsp;&nbsp;<span className="event-datetime">{ this.props.event.date }&nbsp;{ this.props.event.time }</span>
                                <br />
                                { this.props.event.description }
                                <br />
                                { this.props.event.hashtags }
                            </div>
                        </div>
                    </div>

                    { localStorage.getItem( "active" ) === "user" &&
                        <div className="col-3">
                            <button type="button" className="btn red-button" onClick={ this.registerForEvent } hidden={ this.props.registered }>Register</button>
                        </div>
                    }
                    {
                        localStorage.getItem( "active" ) === "restaurant" &&
                        <div className="col-3">
                            { this.props.event.user_names ?
                                this.props.event.user_names.map( ( user_name, index ) => {
                                    return <button type="button" className="btn btn-secondary" onClick={ () => this.showUserProfile( this.props.event.user_ids[ index ] ) }>{ user_name }</button>
                                } )
                                : null }
                        </div>
                    }
                </div>
            </div>
        )
    }
}
