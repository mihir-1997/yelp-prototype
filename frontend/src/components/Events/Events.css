.create-event-form-wrapper {
    background-color: lightgray;
    padding: 4%;
}

.events-wrapper {
    margin: 0 1% 0 10%;
}

.event {
    margin: 10px 0px; 
    padding-top: 10px;
    border-top: 1px lightgray solid;
}

.post-event {
    width: 100%;
    margin: 2% 0 2% 0;
}

.right-align {
    float: right;
}

.post-event-wrapper {
    /* background-color: lightgray; */
}   

.unregistered-event {
    border-right: 1px lightgray solid;
}

.search-event {
    width: 100%;
    margin: 4% 0 2% 0;
}

.event-name,
.event-datetime {
    color: #0073bb;
}

.populer-events,
.registered-events {
    margin-top: 20px;
    color: #d32323;
}

.single-restaurant-event {
    width: 100%;
}