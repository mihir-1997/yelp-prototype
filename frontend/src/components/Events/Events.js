import React, { Component } from 'react'
import axios from 'axios'

import Event from './Event'
import { BACKEND_URL, BACKEND_PORT } from '../Config/backendConfig'

export default class Events extends Component {

    constructor( props ) {
        super( props )
        this.state = {
            events: [],
            unregistered_events: [],
            filtered_unregistered_events: [],
            registered_events: [],
        }
    }

    componentDidMount () {
        let active = localStorage.getItem( "active" )
        if ( active === "restaurant" ) {
            let id = localStorage.getItem( "id" )
            if ( id ) {
                return axios.get( BACKEND_URL + ":" + BACKEND_PORT + "/eventsForRestaurants/" + id )
                    .then( ( res ) => {
                        if ( res.status === 200 ) {
                            // console.log( res.data )
                            this.setState( {
                                events: res.data
                            } )
                        }
                    } )
                    .catch( ( err ) => {
                        if ( err.response ) {
                            console.log( err.response.message )
                            return
                        }
                        return
                    } )
            } else {
                console.log( "No Id found in local storage" )
            }
        } else if ( active === "user" ) {
            let id = localStorage.getItem( "id" )
            if ( id ) {
                return axios.get( BACKEND_URL + ":" + BACKEND_PORT + "/getAllEvents/" + id )
                    .then( ( res ) => {
                        if ( res.status === 200 ) {
                            this.setState( {
                                unregistered_events: res.data.unregistered_events,
                                filtered_unregistered_events: res.data.unregistered_events,
                                registered_events: res.data.registered_events
                            } )
                        }
                    } )
                    .catch( ( err ) => {
                        if ( err.response ) {
                            console.log( err.response.message )
                            return
                        }
                        return
                    } )
            } else {
                console.log( "No Id found in local storage" )
            }
        }
    }

    postEvent = ( item ) => {
        window.location.assign( '/createevent' )
    }

    searchEvent = ( item ) => {
        if ( item.target.value ) {
            this.setState( {
                filtered_unregistered_events: this.state.filtered_unregistered_events.filter( event => event.name.toUpperCase().startsWith( item.target.value.toUpperCase() ) )
            } )
        } else {
            this.setState( {
                filtered_unregistered_events: this.state.unregistered_events
            } )
        }
    }

    showUserProfile = ( user_id ) => {
        console.log( user_id )
        return this.props.history.push( {
            pathname: '/userprofile',
            state: {
                user_id: user_id,
            }
        } )
    }

    render () {
        return (
            <div>
                {
                    localStorage.getItem( "active" ) === "restaurant" &&
                    <div className="container">
                        <div className="row post-event-wrapper">
                            <div className="post-event">
                                <button type="button" className="btn red-button right-align" onClick={ this.postEvent }>Create an Event</button>
                            </div>
                        </div>
                        <div className="row">
                            <div className="col-9 populer-events"><h3>Events</h3></div>
                            <div className="col-3 populer-events"><h5>Registered Customers</h5></div>
                        </div>
                        <div className="row">
                            {
                                this.state.events ?
                                    this.state.events.map( ( event, index ) => {
                                        return <div className="single-restaurant-event" key={ index }>
                                            <Event event={ event } showprofile={ this.showUserProfile }></Event>
                                        </div>
                                    } )
                                    : "No Events. Post a new"
                            }
                        </div>
                    </div>
                }
                {
                    localStorage.getItem( "active" ) === "user" &&
                    <div className="events-wrapper">
                        <div className="row">
                            <div className="col-9 unregistered-event">
                                <div className="row search-event">
                                    <form>
                                        <input type="text" id="searchEvent" className="form-control search-event" placeholder="Search event by name" onChange={ this.searchEvent } />
                                    </form>
                                </div>
                                <div className="row">
                                    <div className="container">
                                        <h4 className="populer-events">Populer Events</h4>
                                    </div>
                                    {
                                        this.state.filtered_unregistered_events ?
                                            this.state.filtered_unregistered_events.map( event => {
                                                return <Event event={ event } registered={ false }></Event>
                                            } )
                                            : "No Events. Come back later!"
                                    }
                                </div>
                            </div>
                            <div className="col-3">
                                <div className="row search-event">
                                    <h4 className="registered-events">Registered Events</h4>
                                </div>
                                <div className="row">
                                    {
                                        this.state.registered_events ?
                                            this.state.registered_events.map( event => {
                                                return <Event event={ event } registered={ true }></Event>
                                            } )
                                            : "No Events. Register for one!"
                                    }
                                </div>
                            </div>
                        </div>
                    </div>
                }
            </div>
        )
    }
}
