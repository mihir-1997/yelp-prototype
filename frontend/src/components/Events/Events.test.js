import React from 'react';
import { shallow } from 'enzyme';

import Events from './Events'

describe( 'Events', () => {

    let axios = require( "axios" );
    let MockAdapter = require( "axios-mock-adapter" );
    let mock = new MockAdapter( axios );

    const BACKEND_URL = process.env.REACT_APP_BACKEND_URL || "http://localhost"
    const BACKEND_PORT = process.env.REACT_APP_BACKEND_PORT || 3001

    const events = [
        {
            date: "2020-09-30",
            description: "New event",
            hashtags: "#tasty",
            id: 2,
            location: "San Jose",
            name: "MCD Event",
            restaurant_id: 4,
            time: "08:00:00",
            user_ids: [ 10, 1 ],
            user_names: [ "Nishit", "Mihir Patel" ]
        },
        {
            date: "2020-10-01",
            description: "A food festival is an event that features a variety of foods, which are usually available for tasting or purchase. The food highlighted at one of these festivals may be a specific ingredient â€” such as",
            hashtags: "#food #foodfest",
            id: 3,
            location: "Santa Clara",
            name: "Food fest",
            restaurant_id: 4,
            time: "09:00:00",
            user_ids: [],
            user_names: []
        }
    ]

    mock.onGet( BACKEND_URL + ":" + BACKEND_PORT + "/eventsForRestaurants/4" ).reply( 200,
        events
    );

    let component

    beforeEach( () => {
        localStorage.setItem( "active", "restaurant" )
        localStorage.setItem( "id", 4 )
        component = shallow( <Events debug /> );
    } );

    it( 'should render correctly in "debug" mode', () => {
        component.instance()
        expect( component ).toMatchSnapshot();
    } );

    it( 'validate localstorage has values', () => {
        expect( localStorage.getItem( "id" ) ).toEqual( 4 )
        expect( localStorage.getItem( "active" ) ).toEqual( "restaurant" )
    } );

    it( 'render after setting localstorage', async ( done ) => {
        component
            .instance()
            .componentDidMount()
            .then( () => {
                expect( component.state().events ).toEqual( events )
                expect( component.find( 'div.single-restaurant-event' ) ).toHaveLength( 2 )
                done()
            } )
        expect( component ).toMatchSnapshot();
    } );

    it( 'validate localstorage is empty', () => {
        localStorage.removeItem( "id" )
        localStorage.removeItem( "active" )
        expect( localStorage.store ).toEqual( {} )
    } );

} )