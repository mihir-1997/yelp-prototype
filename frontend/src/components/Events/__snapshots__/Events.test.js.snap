// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Events render after setting localstorage 1`] = `ShallowWrapper {}`;

exports[`Events should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;
